src/main/java/org/jabref/gui/preview/PreviewViewer.java

```diff
@@ -224,6 +224,13 @@ public void copyPreviewToClipBoard() {
         clipBoardManager.setContent(content);
     }
 
+    public void copySelectionToClipBoard() {
+        ClipboardContent content = new ClipboardContent();
+        content.putString(getSelectionHtmlContent());
```

Comment:
We had some problems in the past with rich-text copy & past, i.e. if you copy bold text and past it into say Word.

For this to work you also need to use `putHtml` as the other methods above. Moreover, the method getSelectionHtmlContent seems to only return the text, not the html, so a better name would be `getSelectionTextContent`. Getting the html might be a bit more difficult, https://stackoverflow.com/questions/4176923/html-of-selected-text/4177234#4177234 is a starting point but maybe there are also simpler ways in the meantime.

Reply:
The JS script above seems to work quite nicely, there is however a downside to it. I don't know if it's intended but it's not extracting html of substrings. For example, if we highlight and copy the title `InProceedings (Salm2020)`, we get `<i>InProceedings</i><a name="Salm2020"> (Salm2020)</a>`, which is correct. However if we only highlight and copy the part `Proceedings` we simply get the plaintext `Proceedings`.

I played around copy & pasting into open office and apart from the limitation above, it works as intended.

Reply:
Thanks for trying it out! That's not perfect, but I would say good enough for most purposes.